content/usgs-packages/dataRetrieval_Assessment.Rmd

```diff
@@ -0,0 +1,126 @@
+---
+title: "dataRetrieval - exercises"
+date: "9999-10-15"
+author: "Lindsay R. Carr"
+slug: "dataRetrieval-exercises"
+image: "img/main/intro-icons-300px/r-logo.png"
+output: USGSmarkdowntemplates::hugoTraining
+parent: Introduction to USGS R Packages
+weight: 3
+draft: true
+---
+
+```{r setup, include=FALSE, warning=FALSE, message=FALSE}
+library(knitr)
+
+knit_hooks$set(plot=function(x, options) {
+  sprintf("<img src='../%s%s-%d.%s'/ title='%s'/>", 
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/dataRetrieval-exercises/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO",
+  message=FALSE
+)
+
+knit_hooks$set(addToggle = function(before, options, envir) {
+    if(before) {
+      sprintf('<button class="ToggleButton" onclick="toggle_visibility(\'%1$s\')">Show Answer</button>
+              <div id="%1$s" style="display:none">', opts_current$get('label'))
+    } else {
+      '</div>'
+    }
+})
+
+set.seed(1)
+```
+
+Before starting the exercises, you should make sure that the `dataRetrieval` package is installed and loaded. If you haven't recently updated, you could reinstall the package by running `install.packages('dataRetrieval')` or go to the "Update" button in the "Packages" tab in RStudio.
+
+```{r load-dataRetrieval}
+# load the dataRetrieval package
+library(dataRetrieval)
+```
+
+## Exercise 1
+
+*Determine the number of sites in Arizona that have lake temperature data available in NWIS. Then find how many Arizona sites have lake temperature data available in WQP.*
+
+```{r addToggle="ON"}
+# NWIS Arizona lake temperature sites
+azlaketemp_nwis <- whatNWISsites(stateCd="AZ", siteType="LK", parameterCd="00010")
+nrow(azlaketemp_nwis)
+
+# WQP Arizona lake temperature sites
+azlaketemp_wqp <- whatWQPsites(statecode="AZ", 
+                               siteType="Lake, Reservoir, Impoundment", 
+                               characteristicName="Temperature, water")
+nrow(azlaketemp_wqp)
+```
+
+## Exercise 2
+
+*Determine which sites in the District of Columbia had daily streamflow below the historic daily average on August 20th, 2013. Hint: use three different functions to figure this out (find site numbers, then statistics data, and then daily value data).*
+
+```{r addToggle="ON"}
+# Find DC site numbers that have streamflow
+dc_stream_sites <- whatNWISsites(stateCd = "DC", siteType="ST", parameterCd="00060")
+
+# Get streamflow from August 20, 2013
+dc_2013_q <- readNWISdv(siteNumbers=dc_stream_sites[['site_no']], parameterCd="00060",
```

Comment:
Alternatively:
```
dc_stream_sites_2 <- readNWISdata(service = "site", stateCd = "DC", 
                                siteType="ST", parameterCd="00060",
                                seriesCatalogOutput=TRUE)
library(dplyr)
dc_stream_sites_2 <- dc_stream_sites_2 %>%
  filter(parm_cd == "00060",
         data_type_cd == "dv")
```
That way, they get to see why only 4 rows of data return when you actually pull data. The fifth only collected data from 1892-08-18 to 1933-09-30!